addons/website_crm/controllers/main.py

```diff
@@ -80,6 +80,12 @@ def insert_record(self, request, model, values, custom, meta=None):
                         values['partner_id'] = visitor_partner.id
                 else:
                     values['partner_id'] = visitor_partner.id
+            elif values_email_normalized and not visitor_partner:
+                email_partner = request.env['res.partner'].sudo().search([('email_normalized', '=', values_email_normalized)], limit=1)
+                email_user = request.env['res.users'].sudo().search([('partner_id', 'in', email_partner.ids)], limit=1)
+                if email_partner and not email_user:
```

Comment:
I guess this is related to https://github.com/odoo/odoo/pull/71794#discussion_r648109549 ?
It might need a comment in the code explaining that briefly so it doesn't get removed later as this is an important check